npm/src/upload/index.ts

```diff
@@ -2,45 +2,49 @@ import { TdrFile } from "@nationalarchives/file-information"
 import { ClientFileProcessing } from "../clientfileprocessing"
 import { ClientFileMetadataUpload } from "../clientfilemetadataupload"
 import { S3Upload } from "../s3upload"
-
-interface HTMLInputTarget extends EventTarget {
-  files?: InputElement
-}
-
-interface InputElement {
-  files?: TdrFile[]
-}
+import { UploadForm } from "./upload-form"
 
 export class UploadFiles {
   clientFileProcessing: ClientFileProcessing
   stage: string
+  goToNextPage: () => void
 
   constructor(
     clientFileProcessing: ClientFileMetadataUpload,
     identityId: string,
-    stage: string
+    stage: string,
+    goToNextPage: () => void
   ) {
     this.clientFileProcessing = new ClientFileProcessing(
       clientFileProcessing,
       new S3Upload(identityId)
     )
     this.stage = stage
+    this.goToNextPage = goToNextPage
   }
 
-  retrieveFiles(target: HTMLInputTarget | null): TdrFile[] {
-    const files: TdrFile[] = target!.files!.files!
-    if (files === null || files.length === 0) {
-      throw Error("No files selected")
+  uploadFiles: (
+    files: TdrFile[],
+    consignmentId: string
+  ) => Promise<void> = async (files: TdrFile[], consignmentId: string) => {
+    const pageUnloadAction: (e: BeforeUnloadEvent) => void = e => {
+      e.preventDefault()
+      e.returnValue = ""
     }
-    return files
-  }
-
-  uploadFilesSuccess(): void {
-    const uploadDataFormRedirect: HTMLFormElement | null = document.querySelector(
-      "#upload-data-form"
-    )
-    if (uploadDataFormRedirect) {
-      uploadDataFormRedirect.submit()
+    window.addEventListener("beforeunload", pageUnloadAction)
+
+    try {
+      await this.clientFileProcessing.processClientFiles(
+        consignmentId,
+        files,
+        this.stage
+      )
+      // In order to prevent exit confirmation when page redirects to Records page
+      window.removeEventListener("beforeunload", pageUnloadAction) // Move to the goToNextPage method?
```

Comment:
Is the comment: "Move to the goToNextPage method?" meant to be here? Little bit confusing whether it is meant to indicate a future enhancement?

Reply:
That is weird, I thought I removed it. I'll do it now; thanks!